xapian-core/weight/lmweight.cc

```diff
@@ -182,12 +202,20 @@ LMWeight::get_maxpart() const
 {
     // Variable to store the collection frequency
     double upper_bound;
+    // Store upper bound on wdf in variable wdf_max
+    double wdf_max = get_wdf_upper_bound();
 
     // Calculating upper bound considering different smoothing option available to user.
     if (select_smoothing == JELINEK_MERCER_SMOOTHING) {
 	upper_bound = (param_smoothing1 * weight_collection) + (1 - param_smoothing1);
     } else if (select_smoothing == DIRICHLET_SMOOTHING) {
-	upper_bound = (get_doclength_upper_bound() + (param_smoothing1 * weight_collection)) / (get_doclength_upper_bound() + param_smoothing1);
+	if (enable_dirplus) {
+	    upper_bound = (1 + (wdf_max/(param_smoothing1 * weight_collection))) *
+			  (1 + (param_delta/(param_smoothing1 * weight_collection)));
+	}
+	else if (!enable_dirplus) {
```

Comment:
This can just be `else` (and should be collapsed with the previous line: `} else {`).